model/jpa/src/main/java/org/keycloak/models/jpa/RealmAdapter.java

```diff
@@ -2345,6 +2345,56 @@ private ComponentEntity getComponentEntity(String id) {
         return c;
     }
 
+    @Override
+    public void patchRealmLocalizationTexts(String locale, Map<String, String> localizationTexts) {
+        Collection<RealmLocalizationTextsEntity> currentLocalizationTexts = realm.getRealmLocalizationTexts();
+        Optional<RealmLocalizationTextsEntity> localizationTextsEntityOptional = currentLocalizationTexts.stream().filter(t -> locale.equals(t.getLocale())).findAny();
+        if(localizationTextsEntityOptional.isPresent()) {
+            RealmLocalizationTextsEntity localizationTextsEntity = localizationTextsEntityOptional.get();
+            localizationTextsEntity.getTexts().putAll(localizationTexts);
+
+            em.persist(localizationTextsEntity);
+        }
+        else {
+            RealmLocalizationTextsEntity realmLocalizationTextsEntity = new RealmLocalizationTextsEntity();
+            realmLocalizationTextsEntity.setRealmId(realm.getId());
+            realmLocalizationTextsEntity.setLocale(locale);
+            realmLocalizationTextsEntity.setTexts(localizationTexts);
+
+            em.persist(realmLocalizationTextsEntity);
+        }
+    }
+
+    @Override
+    public boolean removeRealmLocalizationTexts(String locale) {
+        if (locale == null) return false;
+        Optional<RealmLocalizationTextsEntity> localizationTextsEntity = realm.getRealmLocalizationTexts().stream().filter(t -> locale.equals(t.getLocale())).findAny();
+        if (localizationTextsEntity.isPresent())
+        {
+            em.remove(localizationTextsEntity.get());
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public Map<String, Map<String, String>> getRealmLocalizationTexts() {
+        Map<String, Map<String, String>> localizationTexts = new HashMap<>();
+        realm.getRealmLocalizationTexts().forEach(localizationTextsEntity -> {
+            localizationTexts.put(localizationTextsEntity.getLocale(), localizationTextsEntity.getTexts());
+        });
+        return localizationTexts;
+    }
+
+    @Override
+    public Map<String, String> getRealmLocalizationTextsByLocale(String locale) {
+        Optional<RealmLocalizationTextsEntity> localizationTextsEntity = realm.getRealmLocalizationTexts().stream().filter(l -> locale.equals(l.getLocale())).findAny();
```

Comment:
Since there is no Hibernate cache, any request to `getRealmLocalizationTextsByLocale` would read the `getRealmLocalizationTexts()` in full (all locale), then potentially find one and discard all the rest.
Could this be done in a way that would either only read one localization bundle or always cache all? The latter may affect infinispan layer as well. This is related to the comment in `RealmEntity`.